plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/server/debug/OpenShiftDebugUtils.java

```diff
@@ -176,17 +182,17 @@ private void updateDeploymentConfigValues(IDeploymentConfig deploymentConfig,
 			}
 			if (added) {
 				container.setPorts(ports); 
-				deploymentConfig.setContainers(containers);
+				replicationController.setContainers(containers);
 			}
 		}
 		//TODO the list of env var to set in debug mode should probably be defined in the server settings instead
-		deploymentConfig.setEnvironmentVariable(DEBUG_PORT_KEY, String.valueOf(debugContext.getDebugPort()));
-		deploymentConfig.setEnvironmentVariable(DEV_MODE_KEY, String.valueOf(debugContext.isDebugEnabled()));//for node
-		deploymentConfig.setEnvironmentVariable(DEBUG_KEY, String.valueOf(debugContext.isDebugEnabled()));//for eap
+		replicationController.setEnvironmentVariable(DEBUG_PORT_KEY, String.valueOf(debugContext.getDebugPort()));
+		replicationController.setEnvironmentVariable(DEV_MODE_KEY, String.valueOf(debugContext.isDebugEnabled()));//for node
```

Comment:
@bdshadow can you please take note of this and make sure we take care of it in your refactoring for https://issues.jboss.org/browse/JBIDE-23900 ?